pkg/api/norman/customization/cluster/actions_monitoring.go

```diff
@@ -72,10 +73,11 @@ func (a ActionHandler) editMonitoring(actionName string, action *types.Action, a
 		return httperror.NewAPIError(httperror.InvalidBodyContent, err.Error())
 	}
 
-	cluster = cluster.DeepCopy()
-	cluster.Annotations = monitoring.AppendAppOverwritingAnswers(cluster.Annotations, string(data))
-
-	_, err = a.ClusterClient.Update(cluster)
+	err = updateClusterWithRetryOnConflict(a.ClusterClient, cluster, func(cluster *v3.Cluster) *v3.Cluster {
+		cluster = cluster.DeepCopy()
```

Comment:
I don't think we need to deepcopy objects that are retrieved with client.Get call from etcd/data store.
I think it's only needed for objects from cache

Reply:
updated!